vispy/visuals/axis.py

```diff
@@ -175,64 +173,53 @@ def tick_font_size(self):
     def tick_font_size(self, value):
         self._text.font_size = value
 
-    @property
+    @updating_property
     def tick_direction(self):
-        return self._tick_direction
+        """The tick direction to use (in document coordinates)."""
 
     @tick_direction.setter
-    def tick_direction(self, value):
-        self._tick_direction = np.array(value, float)
-
-    def __setattr__(self, name, val):
-        super().__setattr__(name, val)
-        if self._initialized:
-            if name in (
-                "minor_tick_length",
-                "major_tick_length",
-                "tick_label_margin",
-                "axis_label_margin",
-                "tick_direction",
-                "axis_label",
-                "pos",
-                "scale_type",
-            ):
-                self._need_update = True
-                self.update()
+    def tick_direction(self, tick_direction):
+        self._tick_direction = np.array(tick_direction, float)
 
     @property
     def axis_font_size(self):
-        return self._axis_label.font_size
+        return self._axis_label_vis.font_size
 
     @axis_font_size.setter
     def axis_font_size(self, value):
-        self._axis_label.font_size = value
+        self._axis_label_vis.font_size = value
 
-    @property
-    def axis_label(self):
-        return self._axis_label_text
+    @updating_property
+    def domain(self):
+        """The data values at the beginning and end of the axis, used for tick labels."""
 
-    @axis_label.setter
-    def axis_label(self, axis_label):
-        self._axis_label_text = axis_label
+    @updating_property
+    def axis_label(self):
+        """Text to use for the axis label."""
 
-    @property
+    @updating_property
     def pos(self):
-        return self._pos
+        """Co-ordinates of start and end of the axis."""
 
     @pos.setter
     def pos(self, pos):
         self._pos = np.array(pos, float)
 
-    @property
-    def domain(self):
-        return self._domain
+    @updating_property
+    def minor_tick_length(self):
+        """The length of minor ticks, in pixels"""
```

Comment:
This (an empty method) is probably the worst side effect of this, but I'm still OK with it. Looks odd at first glance though.

Reply:
yeah I agree.  It's definitely the worst part.  looks odd and requires additional research to understand it.

Reply:
the decorator makes it pretty clear that it must be (is the only option as) the culprit so this seems fine